src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c

```diff
@@ -21,26 +27,317 @@ const ActorInit En_Tubo_Trap_InitVars = {
+        func_809308F4(this, globalCtx);
+        func_800F0568(globalCtx, &this->actor.world , 0x28, (u16)0x2802);
+        func_800F0568(globalCtx, &this->actor.world , 0x28, (u16)0x2887);
+        func_8093089C(this, globalCtx);
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    // AT_HIT
+    if ((this->collider.base.flagsAT & 2) != 0) {
+        this->collider.base.flagsAT &= ~2;
+        if (player == this->collider.base.collisionAT) {
+            func_809308F4(this, globalCtx);
+            func_800F0568(globalCtx,  &this->actor.world, 0x28, (u16)0x2887);
+            func_800F0568(globalCtx,  &player2->world, 0x28, (u16)0x83E);
+            func_8093089C(this, globalCtx);
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+    }
+    if (((this->actor.bgCheckFlags & 8) != 0) || ((this->actor.bgCheckFlags & 1) != 0)) {
+        func_809308F4(this, globalCtx);
+        func_800F0568(globalCtx,  &this->actor.world, 0x28, (u16)0x2887);
+        func_8093089C(this, globalCtx);
+        Actor_MarkForDeath(&this->actor);
+    }
+} 
+
+#if NON-MATCHING
+// Wait For Proximity (sitting)
+  // NON-MATCHING: wrong register used, v instead of t for the weirdValues[] pointer
+void func_80931004(EnTuboTrap *this, GlobalContext *globalCtx) {
+    Actor *player = PLAYER; 
+    f32 currentHeight;
+    s8 weirdvalue;
+    s16 startingRotation;
+
+    if ((this->actor.xzDistToPlayer < 200.0f) && (this->actor.world.pos.y <= player->world.pos.y)) {
+        startingRotation = this->actor.home.rot.z;
+        if ((startingRotation == 0) || (this->actor.yDistToPlayer <= ((f32) startingRotation * 10.0f))) {
+            func_800BC154(globalCtx, &globalCtx->actorCtx, this, ACTORCAT_ENEMY);
+            currentHeight = this->actor.world.pos.y;
+            this->actor.flags = |= 0x11; // always update and can target
+            
+            // hard to know what this value is even used for without knowing what ActorPlayer::unk14B is
+            // wild guess: this is player animation state, height is modified to always point at center of link model
+            weirdvalue = D_8093146C[ ((ActorPlayer*) player)->unk14B & 0xFF];
+
+            this->targetY = player->world.pos.y + (f32) weirdvalue;
+            if (this->targetY < currentHeight) {
+                this->targetY = currentHeight;
+                this->targetY += weirdvalue;
+            }
+            this->originPos = this->actor.world.pos;
+            func_800B8EC8(&this->actor, 0x28C4U); // play actorsound2 pot move start
+            this->actionFunc = func_80931138;
+        }
+    }
+} // */
+#else
+  #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/func_80931004.asm")
+#endif
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Tubo_Trap_0x809307E0/EnTuboTrap_Destroy.asm")
+// levitate
+void func_80931138(EnTuboTrap *this, GlobalContext *globalCtx) {
+    this->actor.shape.rot.y += 0x1388;
```

Comment:
```suggestion
    this->actor.shape.rot.y += 5000;
```
Use the most natural number between hex and decimal. Sometimes it's a bit of a guess.

Reply:
Kenix requested in my other PR that all rotation calculations should be left in hex, for now returning to hex.

Feel free to re-request the change if this overrides that.

Reply:
I spoke to Kenix about this, and we agreed it can be decimal, so I'd like to re-request this. Angle calculations are probably almost always in hex, because that is a much more natural to work with them, but that doesn't mean the devs were consistent. So for angles, go with hex unless the decimal number is much rounder, like it is here. Sorry for the back and forth on this!